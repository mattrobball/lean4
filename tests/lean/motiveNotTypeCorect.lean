inductive D : Nat -> Type
def mwe (t : Nat) (f : Nat -> Nat) (h : f t = t)
  (d : D (f t))
  (P : (t : Nat) -> D t -> Prop)
  : P (f t) d
  := by
  rw [h] -- tactic 'rewrite' failed, motive is not type correct
  sorry
