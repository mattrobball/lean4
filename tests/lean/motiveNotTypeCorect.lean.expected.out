motiveNotTypeCorect.lean:7:6-7:7: error: tactic 'rewrite' failed, motive is not type correct
t : Nat
f : Nat → Nat
h : f t = t
d : D (f t)
P : (t : Nat) → D t → Prop
⊢ P (f t) d
motiveNotTypeCorect.lean:18:8-18:9: error: tactic 'rewrite' failed, motive is dependent
h : true = false
A : (b : Bool) → if b = true then Prop else Nat
⊢ A true
